### ISSUE TITLE
realloc going wild in stack_add function

### ISSUE THREAD

**Issue by ctarbide** [kimtg/arcadia#22]
Hi Kim, I am trying to figure out what is happening, maybe you can give me some directions on how to understand what's going on.. consider this:

    echo '((fn (x) (x x)) (fn (x) (x x)))' | ./arcadia

This simple example makes arcadia allocates memory as if there is no end of it, from what I understand the code above should produce an "infinite stream of cons" and run in linear space, my main worry is that this is happening because some design decision that cannot be easily solved. Insights?

The allocation goes berserk in a `realloc` in `stack_add` function, the real cause is unclear to me.

**Comment by kimtg** [comment]
Thank you for your interest. I tested the function, and this problem did not occur in [Arc++](https://github.com/kimtg/Arcpp), SBCL and Racket. This problem occurs because Arcadia uses the C stack and does not have perfect garbage collection. I don't think it will be easily resolved. If this is a problem, please use Arc++, which uses Reference Counting. FYI, the same problem occurs in [newlisp](https://www.newlisp.org/):
```
newLISP v.10.7.5 64-bit on Windows IPv4/6 libffi, options: newlisp -h

> ((fn (x) (x x)) (fn (x) (x x)))

ERR: call or result stack overflow: x
called from user function (x x)
```

**Comment by ctarbide** [comment]
Ok Kim, thanks! Closing the issue now, best regards!

**Comment by kimtg** [comment]
Thanks to your point, the GC bug that occurred when executing ((fn (x) (x x)) (fn (x) (x x))) was fixed in v0.36. Now the previous example program consumes a fixed amount of memory.

**Comment by kimtg** [comment]
Thanks to your point, the GC bug that occurred when executing ((fn (x) (x x)) (fn (x) (x x))) was fixed in v0.36. Now the previous example program consumes a fixed amount of memory.
